bin/elasticurl/main.cpp

```diff
@@ -0,0 +1,540 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/crypto/Hash.h>
+#include <aws/crt/http/HttpConnection.h>
+#include <aws/crt/http/HttpRequestResponse.h>
+#include <aws/crt/io/Uri.h>
+
+#include <aws/common/command_line_parser.h>
+#include <aws/io/uri.h>
+
+#include <condition_variable>
+#include <fstream>
+#include <iostream>
+#include <mutex>
+
+using namespace Aws::Crt;
+
+#define ELASTICURL_VERSION "0.0.1"
+
+std::shared_ptr<Aws::Crt::Io::IStream> input_body = nullptr;
+std::ifstream inputfile;
+
+struct elasticurl_ctx
+{
+    struct aws_allocator *allocator;
+    const char *verb;
+    struct aws_uri uri;
+    bool response_code_written;
+    const char *cacert;
+    const char *capath;
+    const char *cert;
+    const char *key;
+    int connect_timeout;
+    const char *header_lines[10];
+    size_t header_line_count;
+    const char *alpn;
+    bool include_headers;
+    bool insecure;
+    FILE *output;
+    const char *trace_file;
+    Aws::Crt::LogLevel log_level;
+    enum aws_http_version required_http_version;
+    bool exchange_completed;
+    bool bootstrap_shutdown_completed;
+};
+
+static void s_usage(int exit_code)
```

Comment:
I removed all those signing-lib and signing-func things, since I don't know what they are...... Should I put them back?